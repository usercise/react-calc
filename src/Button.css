.Button {
  border-radius: 6px;
  padding: 4px;
}
