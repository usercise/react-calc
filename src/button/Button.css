.Button {
  border-width: 0;
  border-radius: 6px;
  padding: 4px;
  background-color: rgb(221, 146, 34);
  color: #fff;
  line-height: 1.5em;
  font-size: 1.2rem;
  font-weight: 600;
}

.Button.isActive {
  color: rgb(221, 146, 34);
  background-color: #fff;
}

.Button--plus {
  grid-row: 2 / 4;
  grid-column: 4;
}

.Button--equal {
  grid-row: 4 / 6;
  grid-column: 4;
}

.Button--zero {
  grid-column: 1 / 3;
}
