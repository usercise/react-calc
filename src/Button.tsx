import React from 'react';
import './Button.css';

interface ButtonProps {
  title: string;
  callback: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
  buttonValue?: any;
  modifier?: string;
}

const Button: React.SFC<ButtonProps> = props => {
  const classNames = props.modifier
    ? `Button Button--${props.modifier}`
    : 'Button';
  return (
    <button
      className={classNames}
      type="button"
      value={props.buttonValue}
      onClick={props.callback}
    >
      {props.title}
    </button>
  );
};

export default Button;
