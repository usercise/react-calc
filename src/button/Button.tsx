import React from 'react';
import './Button.css';

interface ButtonProps {
  title: string;
  callback: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
  buttonValue?: any;
  modifier?: string;
  isActive?: boolean;
}

const Button: React.SFC<ButtonProps> = props => {
  const isActive = props.isActive ? ' isActive' : '';
  const classNames = props.modifier
    ? `Button${isActive} Button--${props.modifier}`
    : `Button${isActive}`;
  return (
    <button
      className={classNames}
      type="button"
      value={props.buttonValue}
      onClick={props.callback}
    >
      {props.title}
    </button>
  );
};

export default Button;
