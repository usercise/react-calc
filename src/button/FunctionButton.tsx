import React from 'react';
import Button from './Button';

interface FunctionButtonProps {
  call: string;
  displayValue: string;
  modifier?: string;
  functionButtonUpdateValue: (newFunction: string, isEditing: boolean) => void;
}

const FunctionButton: React.SFC<FunctionButtonProps> = props => {
  const buttonCall = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
    props.functionButtonUpdateValue(props.call, false);
  };
  return (
    <Button
      modifier={props.modifier}
      title={props.call}
      callback={buttonCall}
    />
  );
};
export default FunctionButton;
