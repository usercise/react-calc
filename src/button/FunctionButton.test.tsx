import React from 'react';
import { mount, shallow, render } from 'enzyme';
import Button from './Button';
import FunctionButton from './FunctionButton';

describe('<FunctionButton />', () => {
  it('renders without crashing', () => {
    expect(
      shallow(
        <FunctionButton call="/" functionButtonUpdateValue={() => {}} />
      ).find(Button)
    ).toHaveLength(1);
  });

  it('Button DOM to not have changed', () => {
    const component = shallow(
      <FunctionButton call="/" functionButtonUpdateValue={() => {}} />
    );
    expect(component.html()).toMatchSnapshot();
  });

  it('modifier gets set', () => {
    const wrapper = shallow(
      <FunctionButton
        call="/"
        functionButtonUpdateValue={() => {}}
        modifier="test"
      />
    );
    expect(wrapper.find(Button).prop('modifier')).toEqual('test');
  });

  it('title gets set', () => {
    const wrapper = shallow(
      <FunctionButton
        call="/"
        functionButtonUpdateValue={() => {}}
        modifier="test"
      />
    );
    expect(wrapper.find(Button).prop('title')).toEqual('/');
  });

  it('simulates click events', () => {
    const onButtonClick = jest.fn();
    const wrapper = mount(
      <FunctionButton call="/" functionButtonUpdateValue={onButtonClick} />
    );
    wrapper.find('button').simulate('click');
    expect(onButtonClick).toHaveBeenCalled();
  });

  it('click events has currect parameter', () => {
    const onButtonClick = jest.fn();
    const wrapper = mount(
      <FunctionButton call="/" functionButtonUpdateValue={onButtonClick} />
    );
    wrapper.find('button').simulate('click');
    expect(onButtonClick).toHaveBeenCalledWith('/', false);
  });
});
