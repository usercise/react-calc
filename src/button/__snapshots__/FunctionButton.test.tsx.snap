// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FunctionButton /> Button DOM to not have changed 1`] = `"<button class=\\"Button\\" type=\\"button\\">/</button>"`;
